astropy/io/misc/hdf5.py

```diff
@@ -197,21 +197,6 @@ def _encode_mixins(tbl):
     from astropy import units as u
     from astropy.utils.data_info import MixinInfo, serialize_context_as
 
-    # If PyYAML is not available then check to see if there are any mixin cols
-    # that *require* YAML serialization.  HDF5 already has support for
-    # Quantity, so if those are the only mixins the proceed without doing the
-    # YAML bit, for backward compatibility (i.e. not requiring YAML to write
-    # Quantity).
-    try:
-        import yaml  # noqa
-    except ImportError:
-        for col in tbl.itercols():
-            if (has_info_class(col, MixinInfo) and
-                    col.__class__ is not u.Quantity):
-                raise TypeError("cannot write type {} column '{}' "
-                                "to HDF5 without PyYAML installed."
-                                .format(col.__class__.__name__, col.info.name))
-
```

Comment:
lint: since this change `has_info_class` and `MixinInfo` are no longer used, so those imports can be removed.

Reply:
Done!